### PR TITLE
[DDW-430] Smart tooltips

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR implements Smart Tooltips component across whole Daedalus application

## Screenshots

Screen Shot 2020-11-21 at 11 33 54 AM

Screen Shot 2020-11-21 at 11 34 10 AM

Screen Shot 2020-11-21 at 11 35 19 AM

Screen Shot 2020-11-21 at 11 35 33 AM

Screen Shot 2020-11-21 at 11 35 46 AM

Screen Shot 2020-11-21 at 11 35 53 AM

Screen Shot 2020-11-21 at 11 36 05 AM

Screen Shot 2020-11-21 at 11 36 17 AM

Screen Shot 2020-11-21 at 11 36 30 AM

Screen Shot 2020-11-21 at 11 37 10 AM

Screen Shot 2020-11-21 at 11 37 30 AM

Screen Shot 2020-11-21 at 11 38 03 AM

Screen Shot 2020-11-21 at 11 40 44 AM

Screen Shot 2020-11-21 at 11 40 56 AM

Screen Shot 2020-11-21 at 11 41 10 AM

Screen Shot 2020-11-21 at 11 41 23 AM

Screen Shot 2020-11-21 at 11 41 42 AM

Screen Shot 2020-11-21 at 11 41 50 AM

Screen Shot 2020-11-21 at 11 44 29 AM

Screen Shot 2020-11-21 at 11 44 36 AM

Screen Shot 2020-11-21 at 11 44 44 AM

Screen Shot 2020-11-21 at 11 44 58 AM

Screen Shot 2020-11-21 at 12 34 35 PM

Screen Shot 2020-11-21 at 12 35 04 PM

Screen Shot 2020-11-21 at 12 42 57 PM

Screen Shot 2020-11-21 at 12 49 40 PM

Screen Shot 2020-11-21 at 12 53 56 PM

Screen Shot 2020-11-21 at 12 54 09 PM

Screen Shot 2020-11-21 at 12 54 48 PM

Screen Shot 2020-11-21 at 12 55 02 PM

Screen Shot 2020-11-21 at 12 55 14 PM

Screen Shot 2020-11-21 at 1 08 12 PM

Screen Shot 2020-11-21 at 1 07 54 PM

---

## Testing Checklist

- [Slack QA thread]()
- Check the following tooltips:
- [x] Connecting/Loading Screen - All Status Icons
- [x] Syncing status
- [x] Create Wallet Wizard - Enter Password Input
- [x] Restore Wallet Wizard -  Enter Password Input
- [x] Byron Wallet Settings - Paper Wallet Disclaimer
- [x] Wallet Settings - Change Password - Current Password Input
- [x] Daedalus Diagnostics - Daedalus State Directory
- [x] Daedalus Diagnostics - CPU
- [x] Stake Pool Details - All listed items
- [x] Stake Pool Details - Stake Pool ID
- [x] Delegation center - Delegated wallet
- [x] Stake Pools View - Saturation bar values

## Testing Summary

For build 15243:

**New**:
1 - Tooltips with long text are too wide. Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970).

**Rejected:**
~1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) - Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)~

**Fixed**
1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
2 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).`
3 - The pool id component moves for an instant after copying the value. [Check this recording](https://drive.google.com/file/d/1lMUy-2rMkdrbIgmYP7Pwd3oQPPq5sldC/view?usp=sharing). - [Reported here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-729734017)
4 -When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password.  [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) - Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
5 - The issue reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) about the ? icon and the tooltip (New # 2) still reproduces when restoring a wallet. Check [screenshot](https://drive.google.com/file/d/1SQTTCaavLdhy_FyVaKzgZPk1UkL53FqB/view?usp=sharing).

---

### Basics

- [x] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [x] In English
  - [x] In Japanese
- [x] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [x] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn flow:test`)
- [x] There are no *lint* errors or warnings (`yarn lint`)
- [x] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [x] Important parts of the code are properly commented and documented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-renderings
- [x] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by gnpf** [comment]
@DeeJayElly here a list of some issues on build 15182:

1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).

2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)

**Comment by DeeJayElly** [comment]
> @DeeJayElly here a list of some issues on build 15182:
> 
> 1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).
> 
> 2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)
> 
> I added to the description of this PR a list of the tooltips to check @mioriohk @ManusMcCole

2. Issue is fixed.

Screen Shot 2020-11-13 at 10 32 38 PM

**Comment by DeeJayElly** [comment]
> @DeeJayElly here a list of some issues on build 15182:
> 
> 1 - On stake pools, the arrow of the tooltip is not over the **?** icon. [This is how it looks now](https://drive.google.com/file/d/1TWT-C5dbFrWJBBN3X15Mt59dVKGg2Wjp/view?usp=sharing), and [this is how it looked before](https://drive.google.com/file/d/1R0MMxFg7GtQi7xG08DGP-tC75OyMWICO/view?usp=sharing).
> 
> 2 - Similar to the previous issue the tooltip on "pool id" is not centered, but the component containing the id is also affected. [Here's the current implementation](https://drive.google.com/file/d/1fCZeehZD_cjO3WWCK5rjPN4Kmgfo6A1L/view?usp=sharing) and [here a screenshot of production.](https://drive.google.com/file/d/1n4lMuIpgGZYu57hYf7hl5M-MFgU2aXN_/view?usp=sharing)
> 
> I added to the description of this PR a list of the tooltips to check @mioriohk @ManusMcCole

1. Issue fixed

**Comment by nikolaglumac** [comment]
@gnpf please re-check 🙏

**Comment by nikolaglumac** [comment]
@DominikGuzei please review this one 🙏

**Comment by mioriohk** [comment]
Hi @DeeJayElly, this is the summary for build 15186

For fixed issues, I have moved them to the top of this PR.

**New issue**
1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing)
 
**Suggestion**
1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? Also occurred during Create Wallet, Restore Wallet and Change password [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing)

**Comment by DeeJayElly** [comment]
> Hi @DeeJayElly, this is the summary for build 15186
> 
> For fixed issues, I have moved them to the top of this PR.
> 
> **New issue**
> 1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing)
> 
> **Suggestion**
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing)

@DominikGuzei do we have tabIndex={-1} implementation inside PopOver component? I believe that needs to be added.

File: source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js
Comment (DominikGuzei) [inline source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js:118]: small thing but here it would be clearer to remove the negation: `isClearTooltipOpeningDownward ? 'bottom': 'top`

File: source/renderer/app/components/staking/widgets/TooltipPool-copyId-tooltip.scss
Comment (DominikGuzei) [inline source/renderer/app/components/staking/widgets/TooltipPool-copyId-tooltip.scss:10]: Can you provide some context why this was necessary? I'm just asking because there is also a Tippy prop for specifying the offsets in JS: https://atomiks.github.io/tippyjs/v6/all-props/#offset

File: source/renderer/app/components/wallet/WalletCreateDialog.scss
Comment (DominikGuzei) [inline source/renderer/app/components/wallet/WalletCreateDialog.scss:82]: ☝️ is this css necessary for the tooltip? it would be great to get rid of any unnecessary css overrides as they are hard to understand

File: source/renderer/app/components/wallet/WalletRestoreDialog.scss
Comment (DominikGuzei) [inline source/renderer/app/components/wallet/WalletRestoreDialog.scss:108]: Same here, you can check https://atomiks.github.io/tippyjs/v6/all-props for available props that might be used instead

File: source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.scss
Comment (DominikGuzei) [inline source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.scss:47]: Same here, position of the tooltip should be done by the PopOver (and adjusted via Tippy props if necessary) - of course if there is a special case im not yet familiar with, please show me :)

File: source/renderer/app/components/wallet/wallet-restore/ConfigurationDialog.scss
Comment (DominikGuzei) [inline source/renderer/app/components/wallet/wallet-restore/ConfigurationDialog.scss:41]: Same here, would prefer to ged rid of any css overrides

**Review by DominikGuzei — CHANGES_REQUESTED** [review]
Great work Alex 🎉 the code looks good apart from some questions regarding CSS overrides for some of the tooltips - i left you a bunch of comments. I haven't debugged the specifics but it would be great if we could ged rid of all css overrides if possible. Let's discuss on Slack!

File: source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js
Comment (DeeJayElly) [reply source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js:118]: Fixed

File: source/renderer/app/components/staking/widgets/TooltipPool-copyId-tooltip.scss
Comment (DeeJayElly) [reply source/renderer/app/components/staking/widgets/TooltipPool-copyId-tooltip.scss]: Removed, its obsolete now.

**Comment by gnpf** [comment]
@DeeJayElly The issues I reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727025104) are fixed (I'm updating the testing summary for this PR on the description).
I also added new elements on the testing checklist @mioriohk @ManusMcCole 

Here's a summary of new and pending issues for build 15200:

**New**
1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
2 - The `?` Icon and the tooltip are overlapping the text in Japanese. [See screenshot](https://drive.google.com/file/d/1HdE-kBe4HZVfetH7BBrP4BigFrIfkLx4/view?usp=sharing).
3 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).

**Pending**
1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password.  [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
 
**Suggestion**
1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)

File: source/renderer/app/components/wallet/wallet-restore/ConfigurationDialog.scss
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/wallet-restore/ConfigurationDialog.scss:33]: Fixed

File: source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.scss
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.scss:39]: Fixed

File: source/renderer/app/components/wallet/WalletRestoreDialog.scss
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/WalletRestoreDialog.scss:83]: Fixed

File: source/renderer/app/components/wallet/WalletCreateDialog.scss
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/WalletCreateDialog.scss:52]: Fixed

**Comment by DeeJayElly** [comment]
> @DeeJayElly The issues I reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727025104) are fixed (I'm updating the testing summary for this PR on the description).
> I also added new elements on the testing checklist @mioriohk @ManusMcCole
> 
> Here's a summary of new and pending issues for build 15200:
> 
> **New**
> 1 - On the “pool id” element, now [you can see a pink border around the component](https://drive.google.com/file/d/1h0lyG9_EBoazE8yb0eR-1z5D_h0mSRNp/view?usp=sharing). This reproduces on other elements after clicking too ([See this slider example](https://drive.google.com/file/d/13yqbOSjyL0jRCqEFow3pMEUOpn-04NQw/view?usp=sharing)) and I think it can be related to the issue Mior reported before (see pending issues)
> 2 - The `?` Icon and the tooltip are overlapping the text in Japanese. [See screenshot](https://drive.google.com/file/d/1HdE-kBe4HZVfetH7BBrP4BigFrIfkLx4/view?usp=sharing).
> 3 - The tooltip for the state directory has too much empty space and is displayed on two rows now. [See screenshot](https://drive.google.com/file/d/16NHLj2oDpNFrgIxdzADr6u5bzcGhp4Yy/view?usp=sharing).
> 
> **Pending**
> 1 - When I tabbed across fields, the smart tooltip icon was also selected during tabbing. This did not exist in production. Also occurred during Create Wallet, Restore Wallet and Change password. [See here](https://drive.google.com/file/d/1-OYVF0sUIMaYxcfGH5ZYX4PVbTm6IIVt/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)
> 
> **Suggestion**
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? [See here](https://drive.google.com/file/d/1jvAv7ACwneoy7X1bBsEUiY2HWJVjf1vT/view?usp=sharing) ///Reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587)

Hi @gnpf @mioriohk @ManusMcCole all the requested changes and fixes have been applied. Pls retest it.

**Review by DominikGuzei — APPROVED** [review]
Amazing cleanup, thanks! 🙏 code looks good now 🎉

**Comment by gnpf** [comment]
@DeeJayElly for build 15215:

**New**:
1 - There pool id component moves for an instant after copying the value. [Check this recording](https://drive.google.com/file/d/1lMUy-2rMkdrbIgmYP7Pwd3oQPPq5sldC/view?usp=sharing).

**Reincident**:
1 - The issue reported [here](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) about the ? icon and the tooltip (New # 2) still reproduces when restoring a wallet. Check [screenshot](https://drive.google.com/file/d/1SQTTCaavLdhy_FyVaKzgZPk1UkL53FqB/view?usp=sharing).

**Pending:**
1 - From [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) `pending issue # 1 `and `suggestion # 1`

**Fixed**
1 - 1 - From [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-728292677) `new issue # 1` and `new issue # 3`

**Review by DominikGuzei — CHANGES_REQUESTED** [review]
I just realized that this PR is using an older version of react-polymorph (that's also why the tabbing issue exists, which i already fixed in the latest version).

So please update to 0.9.7-rc.5 and double-check the width of all tooltips you upgraded, they all need a `maxWidth` (pixel as number) prop otherwise they will stretch over the full width by default

**Comment by DeeJayElly** [comment]
> I just realized that this PR is using an older version of react-polymorph (that's also why the tabbing issue exists, which i already fixed in the latest version).
> 
> So please update to 0.9.7-rc.5 and double-check the width of all tooltips you upgraded, they all need a `maxWidth` (pixel as number) prop otherwise they will stretch over the full width by default

Updated @DominikGuzei

**Comment by DominikGuzei** [comment]
@DeeJayElly you updated to latest react-polymorph but forgot to add the `maxWidth` props to all tooltips with long content like here:
grafik

**Comment by gnpf** [comment]
@DeeJayElly For build **15243** I updated the summary of this PR.
Besides [the issue with the widht of the tooltips](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970) mentioned by @DominikGuzei there are now new incidences.

Would this suggestion reported by @mioriohk be addressed on this PR? Maybe this needs to be discussed @nikolaglumac ?

> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? See here

It was reported on [this comment](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-727907587).

**Comment by nikolaglumac** [comment]
> 1 - When testing: Byron Wallet Settings - Paper Wallet Disclaimer. I noticed the tooltip didn't exactly appear directly above the hovered element. I could replicate this in the production but I feel for consistency with the others eg: "Password field", perhaps this should also display the same way? See here

No, this should remain as is. This is a different case and it has been implemented according to the original design.

**Comment by nikolaglumac** [comment]
The only issue which remains here is the one @DominikGuzei reported in https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970

cc @DeeJayElly @gnpf

**Comment by DeeJayElly** [comment]
> @DeeJayElly you updated to latest react-polymorph but forgot to add the `maxWidth` props to all tooltips with long content like here:
> grafik

Added

**Comment by DeeJayElly** [comment]
> The only issue which remains here is the one @DominikGuzei reported in [#2243 (comment)](https://github.com/input-output-hk/daedalus/pull/2243#issuecomment-730476970)
> 
> cc @DeeJayElly @gnpf

Fixed @gnpf @nikolaglumac @DominikGuzei

**Review by ManusMcCole — APPROVED** [review]

**Comment by gnpf** [comment]
@DeeJayElly All tooltips look good, except the [Daedalus Paper Wallet Disclaimer](https://drive.google.com/file/d/19-9nEL6MmP0AKC2W1nWSTFy9DhUBDkik/view?usp=sharing) (Text is out of the component)

**Comment by DeeJayElly** [comment]
> @DeeJayElly All tooltips look good, except the [Daedalus Paper Wallet Disclaimer](https://drive.google.com/file/d/19-9nEL6MmP0AKC2W1nWSTFy9DhUBDkik/view?usp=sharing) (Text is out of the component)

Check it now @gnpf , it should be ok now.

**Comment by nikolaglumac** [comment]
@gnpf please test again 🙏

**Review by gnpf — APPROVED** [review]
All tooltips look good, great job @DeeJayElly

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@DeeJayElly @a-rukin @gnpf how can this one be approved - don't you see the line-height of the tooltip content? 😢 

Screenshot 2020-11-20 at 17 40 38

**Comment by a-rukin** [comment]
https://zpl.io/2p5dz9l - here you can find a proper line height for the tooltip (15px)

**Comment by a-rukin** [comment]
Font size - 14px 
Line height - 19px

**Comment by nikolaglumac** [comment]
@DeeJayElly please check **every single tooltip** and post screenshots of the fixed state for each of them both in English and Japanese. The font size and line-heights should be as @a-rukin stated. Thanks!

**Comment by DeeJayElly** [comment]
> Font size - 14px
> Line height - 19px

Added

**Review by nikolaglumac — CHANGES_REQUESTED** [review]
@DeeJayElly please add screenshots of every single tooltip affected by this PR in both English and Japanese.

**Comment by DeeJayElly** [comment]
> @DeeJayElly please add screenshots of every single tooltip affected by this PR in both English and Japanese.

Added

**Review by mioriohk — APPROVED** [review]
Looks good to me 👍

**Comment by a-rukin** [comment]
It seems to me that this distance is bigger (+ arrow shifted to the left)
image

**Comment by DeeJayElly** [comment]
> It seems to me that this distance is bigger (+ arrow shifted to the left)
> image

Fixed @a-rukin

**Comment by a-rukin** [comment]
image
it is still shifted to the left

**Review by a-rukin — APPROVED** [review]
ok

**Review by DominikGuzei — APPROVED** [review]
Code is looking good now 👍

**Comment by DominikGuzei** [comment]
Great work @DeeJayElly 🎉 let's wait for final approval by @nikolaglumac

File: source/renderer/app/themes/overrides/PopOverOverrides.scss
Comment (nikolaglumac) [inline source/renderer/app/themes/overrides/PopOverOverrides.scss:2]: @DominikGuzei isn't there a better way to target the content and set up the line-height?

Comment (DominikGuzei) [reply source/renderer/app/themes/overrides/PopOverOverrides.scss:2]: @nikolaglumac good point, we could probably 1. add a sane default in react-polymorph and 2. make it configurable via theme variable 👍 I also saw that most of the `maxWidth` values are 350, so I will also add that as default

Comment (nikolaglumac) [reply source/renderer/app/themes/overrides/PopOverOverrides.scss:2]: Will you take care of that?

**Review by nikolaglumac — APPROVED** [review]

**Pull request merged**
